projects/firefox/build.sh

```diff
@@ -0,0 +1,183 @@
+# output manually filtered for libraries which are available on base-runner.
+# Patching even some libraries out doesn't work (undefined symbol errors).
+REQUIRED_LIBRARIES=(
+  libasyncns.so.0
+  libatk-1.0.so.0
+  libatk-bridge-2.0.so.0
+  libatspi.so.0
+  libboost_filesystem.so.1.58.0
+  libboost_system.so.1.58.0
+  libcairo-gobject.so.2
+  libcairo.so.2
+  libcapnp-0.5.3.so
+  libdatrie.so.1
+  libdbus-1.so.3
+  libdbus-glib-1.so.2
+  libepoxy.so.0
+  libffi.so.6
+  libFLAC.so.8
+  libfontconfig.so.1
+  libfreetype.so.6
+  libgdk-3.so.0
+  libgdk_pixbuf-2.0.so.0
+  libgio-2.0.so.0
+  libglib-2.0.so.0
+  libgmodule-2.0.so.0
+  libgobject-2.0.so.0
+  libgraphite2.so.3
+  libgthread-2.0.so.0
+  libgtk-3.so.0
+  libharfbuzz.so.0
+  libICE.so.6
+  libjson-c.so.2
+  libkj-0.5.3.so
+  libmirclient.so.9
+  libmircommon.so.7
+  libmircore.so.1
+  libmirprotobuf.so.3
+  libogg.so.0
+  libpango-1.0.so.0
+  libpangocairo-1.0.so.0
+  libpangoft2-1.0.so.0
+  libpixman-1.so.0
+  libpng12.so.0
+  libprotobuf-lite.so.9
+  libpulse.so.0
+  libpulsecommon-8.0.so
+  libSM.so.6
+  libsndfile.so.1
+  libthai.so.0
+  libvorbis.so.0
+  libvorbisenc.so.2
+  libwayland-client.so.0
+  libwayland-cursor.so.0
+  libwayland-egl.so.1
+  libwrap.so.0
+  libX11-xcb.so.1
+  libX11.so.6
+  libXau.so.6
+  libxcb-render.so.0
+  libxcb-shm.so.0
+  libxcb.so.1
+  libXcomposite.so.1
+  libXcursor.so.1
+  libXdamage.so.1
+  libXdmcp.so.6
+  libXext.so.6
+  libXfixes.so.3
+  libXi.so.6
+  libXinerama.so.1
+  libxkbcommon.so.0
+  libXrandr.so.2
+  libXrender.so.1
+  libXt.so.6
+)
+
+mkdir $WORK/APT
+chown _apt $WORK/APT # suppress warning message on each file
+cd $WORK/APT
+
+# Download packages which have the required library files.
+# Note that apt-file is very slow, hence parallel is used.
+PACKAGES=($(parallel apt-file search -lx "{}$" ::: ${REQUIRED_LIBRARIES[@]}))
+for PACKAGE in ${PACKAGES[@]}
+do
+  if [[ $PACKAGE =~ dev$ ]]; then continue; fi
+  if [[ $PACKAGE =~ dbg$ ]]; then continue; fi
+  apt-get -q download $PACKAGE
+done
+
+mkdir TMP
+# Extract downloaded packages.
+find -name \*.deb -exec dpkg --extract "{}" TMP \;
+
+mkdir $OUT/lib
+# Move required libraries (and symlinks) to LD_LIBRARY_PATH set below.
+for LIBRARY in ${REQUIRED_LIBRARIES[@]}
+do
+  find TMP -name $LIBRARY\* -exec mv "{}" $OUT/lib \;
+done
+
+# Build a wrapper binary for each target to set environment variables.
+for FUZZ_TARGET in ${FUZZ_TARGETS[@]}
+do
+cat << EOF > $FUZZ_TARGET.c
+#include <stdlib.h>
+#include <string.h>
+#include <unistd.h>
+int main(int argc, char* argv[]) {
+  setenv("MOZ_RUN_GTEST", "1", 1);
+  setenv("LIBFUZZER", "1", 1);
+  setenv("FUZZER", "${FUZZ_TARGET}", 1);
+  setenv("LD_LIBRARY_PATH", "${OUT}/lib", 0); // not set on base-runner
+  // Temporary (or permanent?) work-around for a bug in the fuzzing interface.
+  // https://bugzilla.mozilla.org/show_bug.cgi?id=1466021#c9
+  char* options = getenv("ASAN_OPTIONS");
+  strcat(options, ":detect_stack_use_after_return=0");
```

Comment:
I don't think this is safe at all

Reply:
You're right. I'm surprised now that it even works. Probably overwrites an unimportant variable name.